nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Discover/SdkProjectDiscovery.cs

```diff
@@ -304,8 +312,59 @@ runtimePackageVersion is null ||
             }
         }
 
+        if (requiresManualPackageResolution)
+        {
+            // we were able to collect all <PackageReference> elements, but no transitive dependencies were resolved
+            // to do this we create a temporary project with all of the top-level project elements, resolve _again_, then rebuild the proper result
+            packagesPerProject = await RebuildPackagesPerProject(
+                repoRootPath,
+                workspacePath,
+                startingProjectPath,
+                tfms,
+                packagesPerProject,
+                explicitPackageVersionsPerProject,
+                packagesReplacedBySdkPerProject,
+                topLevelPackagesPerProject,
+                resolvedProperties,
+                referencedProjects,
+                importedFiles,
+                additionalFiles,
+                experimentsManager,
+                logger
+            );
+        }
+
         // and done
-        var projectDiscoveryResults = packagesPerProject.Keys.OrderBy(p => p).Select(projectPath =>
+        var projectDiscoveryResults = BuildResults(
+            repoRootPath,
+            workspacePath,
+            packagesPerProject,
+            explicitPackageVersionsPerProject,
+            packagesReplacedBySdkPerProject,
+            topLevelPackagesPerProject,
+            resolvedProperties,
+            referencedProjects,
+            importedFiles,
+            additionalFiles
+        );
+        return projectDiscoveryResults;
+    }
+
+    private static ImmutableArray<ProjectDiscoveryResult> BuildResults(
```

Comment:
This behavior used to be inline, but it's nicer to look at as a separate function.  No change, just copy/paste.